### ISSUE TITLE
Position issue when Create a multi selection programmatic 

### ISSUE THREAD

**Issue by Sunilthecoder007** [fabricjs/fabric.js#5035]
I am trying to create a multi selection programmatic  section is crated  but newly created slection moves all  objects to outside of canvas below is my function:

**function deSelectLocked() {
    objs = activeCanvas.getActiveObjects();
    var arr = [];
    objs.forEach(function(x) {
        if (x.lock !== true) {
            arr.push(x);
        }
    });
    y = new fabric.ActiveSelection(arr, {
        canvas: activeCanvas
    });
}**

Basically i am trying to deselect  some objects from ActiveSelection created by user on canvas i want to make a selection of  objects which are not locked ..i know that i can easily do this by setting **selectable:false** to object but in my case i want  that if a locked object is clicked then it can be slected  to unlock again but when selecting multiple objects via mouse drag than ignore objects which are already locked

**Comment by asturur** [comment]
i think you should use the removeWithUpdate method from the active selection, i think it should work a bit better.

if you create an active selection from objects that are already in the active selection, you create confusion with the coordinate system.

**Comment by Sunilthecoder007** [comment]
@asturur  how can i use removeWithUpdate  with current active selection

**Comment by asturur** [comment]
currentselection is canvas.getActiveObject, while getActiveObjects gives you the select objects array

**Comment by Sunilthecoder007** [comment]
but removeWithUpdate  is for group i think i need to deselect object from  active selection not a group please show me little code to do that i am confused at this. Thanks

**Comment by asturur** [comment]
Activeselection is a Group indeed.

**Comment by Sunilthecoder007** [comment]
Thank you so much my problem is solved now. :)
